packages/ics721-types/src/token_types.rs

```diff
@@ -0,0 +1,145 @@
+use std::ops::Deref;
+
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Binary, StdResult};
+use cw_storage_plus::{Bound, Bounder, Key, KeyDeserialize, Prefixer, PrimaryKey};
+
+/// A token ID according to the ICS-721 spec. The newtype pattern is
+/// used here to provide some distinction between token and class IDs
+/// in the type system.
+#[cw_serde]
+pub struct TokenId(String);
+
+/// A token according to the ICS-721 spec.
+#[cw_serde]
+pub struct Token {
+    /// A unique identifier for the token.
+    pub id: TokenId,
+    /// Optional URI pointing to off-chain metadata about the token.
+    pub uri: Option<String>,
+    /// Optional base64 encoded metadata about the token.
+    pub data: Option<Binary>,
+}
+
+/// A class ID according to the ICS-721 spec. The newtype pattern is
+/// used here to provide some distinction between token and class IDs
+/// in the type system.
+#[cw_serde]
+pub struct ClassId(String);
+
+#[cw_serde]
+pub struct Class {
+    /// A unique (from the source chain's perspective) identifier for
+    /// the class.
+    pub id: ClassId,
+    /// Optional URI pointing to off-chain metadata about the class.
+    pub uri: Option<String>,
+    /// Optional base64 encoded metadata about the class.
+    pub data: Option<Binary>,
+}
+
+impl TokenId {
+    pub fn new<T>(token_id: T) -> Self
+    where
+        T: Into<String>,
+    {
+        Self(token_id.into())
+    }
+}
+
+impl ClassId {
+    pub fn new<T>(class_id: T) -> Self
+    where
+        T: Into<String>,
+    {
+        Self(class_id.into())
+    }
+}
+
+impl<'a> Bounder<'a> for ClassId {
+    fn inclusive_bound(self) -> Option<cw_storage_plus::Bound<'a, Self>> {
```

Comment:
What exactly is this doing?

Reply:
This is needed in `query_nft_contracts()` since paginated query requires a bounder. Simple need trick for inferring a bounder: https://github.com/arkprotocol/ark-cw-ics721/blob/incoming_proxy/packages/ics721/src/query.rs#L138
Bounder defined here: https://github.com/DA0-DA0/dao-contracts/blob/v2.4.0/packages/cw-paginate-storage/src/lib.rs#L18

Reply:
> 

Well my question is, it seems like the code is just copy paste something that already exists? Doesn't it already get this functionality through an import? What specifically is changing in the override?

Reply:
No, it doesn't exist yet. If I dont implement bounder, then I have to explicitely provide bounder in query.